allennlp/modules/transformer/transformer_encoder.py

```diff
@@ -0,0 +1,99 @@
+from typing import Union, Optional, Dict
+
+import torch
+
+from allennlp.common import FromParams
+
+from allennlp.modules.util import replicate_layers
+from allennlp.modules.transformer.transformer_layer import TransformerLayer
+from allennlp.modules.transformer.transformer_module import TransformerModule
+
+
+class TransformerEncoder(TransformerModule, FromParams):
```

Comment:
I'd call this `TransformerBlock`. This is really a bunch of fully-formed transformer layers stacked together, right? End-to-end BERT would be a `Embedding` plus a `TransformerBlock`?

Reply:
When it's time to write docs for this, we should be clear about which components correspond to which BERT, and how they plug together.